connector/oidc/oidc.go

```diff
@@ -425,6 +431,27 @@ func (c *oidcConnector) createIdentity(ctx context.Context, identity connector.I
 				}
 			}
 		}
+		c.logger.Debugf("Groups::: %q \n", groups)
+
+		// Validate that the user is part of allowedGroups
+		var accessAllowed = false
+		if len(c.allowedGroups) > 0 {
+			for _,group := range c.allowedGroups {
+				if slices.Contains(groups, group) {
+					accessAllowed = true
+					c.logger.Debugf("Found a match - Group::: %q \n", group)
+					break
+				}
+			}
+		} else {
+			// don't check for groups if allowedGroups is not configured
+			c.logger.Debugf("Group check not performed \n")
+			accessAllowed = true
+		}
+
+		if !accessAllowed {
+			return identity, errors.New("User is not in the allowed group(s)")
+		}
```

Comment:
```suggestion
		if len(c.allowedGroups) > 0 {
			groupMatches := groups.Filter(ident.Groups, p.allowedGroups)

			if len(groupMatches) == 0 {
				// No group membership matches found, disallowing
				return identity, fmt.Errorf("user not a member of allowed groups")
			}
			
			groups = groupMatches
		}
```

https://github.com/dexidp/dex/blob/master/pkg/groups/groups.go Should be used here to be inline with other connectors.

Reply:
waiting for this one, because it was lost on rebasing

Reply:
Fixed in https://github.com/dexidp/dex/pull/3074/commits/b3e5bf544bbc192d6013f4b361c597ad32a4bd4c